credentials/tls.go

```diff
@@ -112,6 +116,22 @@ func (c *tlsCreds) ClientHandshake(ctx context.Context, authority string, rawCon
 		conn.Close()
 		return nil, nil, ctx.Err()
 	}
+
+	// The negotiated protocol can be either of the following:
+	// 1. h2: When the server supports ALPN. Only HTTP/2 can be negotiated since
+	//    it is the only protocol advertised by the client during the handshake.
+	//    The tls library ensures that the server chooses a protocol advertised
+	//    by the client.
+	// 2. "" (empty string): If the server doesn't support ALPN. ALPN is a requirement
+	//    for using HTTP/2 over TLS. We can terminate the connection immediately.
+	np := conn.ConnectionState().NegotiatedProtocol
+	if np == "" {
+		if envconfig.EnforceALPNEnabled {
+			_ = conn.Close()
+			return nil, nil, fmt.Errorf("cannot check peer: missing selected ALPN property")
+		}
+		logger.Warning("Allowing TLS connection to server %q with ALPN disabled")
```

Comment:
Also, you are missing the server name in the argument list even though you have the `%q` verb.

Reply:
>Please append something like this to this warning log. TLS connections to servers with ALPN disabled will be disallowed in future grpc-go releases. This will increase the possibility of this being picked up by the user.

Changed the log message. 

> Also, you are missing the server name in the argument list even though you have the %q verb.

Nice catch! Changed to `Warningf` and added the server name.